writeToStream.js

```diff
@@ -520,7 +535,7 @@ function writeString (stream, string) {
  * @api private
  */
 function writeNumber (stream, number) {
-  return stream.write(numCache[number])
+  return stream.write(getNumber(number))
 }
```

Comment:
Not sure I understand you correctly. We anyway need a function call in case of buffer allocation (w/o cache). So `getNumber` is a function only for common interface of getting buffers.